lib/chef/knife/cleanup_versions.rb

```diff
@@ -94,13 +98,53 @@ def cookbooks
         end
         print "\n"
       end
-      
+
       if !config[:delete]
         ui.msg "Not deleting unused cookbook versions; use --delete if you want to remove them"
       end
-      
     end
-    
+
+    def keep_for_runlist(cb_versions, env_name, runlist)
+      runlist_req = { "run_list"  => [ runlist ] }
+      begin
+        run_cookbooks = \
+          rest.post_rest("/environments/#{env_name}/cookbook_versions", runlist_req)
```

Comment:
@pburkholder: trying to figure out what this exactly does. It gives me the following error:
rest.post_rest("/environments/#{env_name}/cookbook_versions", runlist_req)
Net::HTTPServerException: 412 "Precondition Failed"